File: src/path/HolonomicPath.cpp
// Copyright (c) TrajoptLib contributors

#include "path/HolonomicPath.h"

#include <vector>

#include "path/HolonomicWaypoint.h"
#include "path/Path.h"
#include "path/Waypoint.h"

namespace trajopt {

HolonomicPath::HolonomicPath(
    std::vector<HolonomicWaypoint> holonomicWaypoints, const Obstacle& bumpers,
    std::vector<Constraint> globalConstraints,
    std::vector<HolonomicConstraint> globalHolonomicConstraints)
    : Path(bumpers, std::move(globalConstraints)),
      holonomicWaypoints(std::move(holonomicWaypoints)),
      globalHolonomicConstraints(std::move(globalHolonomicConstraints)) {}

size_t HolonomicPath::Length() const noexcept {
  return holonomicWaypoints.size();
}
Waypoint& HolonomicPath::GetWaypoint(size_t index) {
  return holonomicWaypoints[index];
}
const Waypoint& HolonomicPath::GetWaypoint(size_t index) const {
  return holonomicWaypoints[index];
}

}  // namespace trajopt

File: java/src/main/java/org/sleipnirgroup/trajopt/HolonomicPath.java
// Copyright (c) TrajoptLib contributors

package org.sleipnirgroup.trajopt;

import static org.sleipnirgroup.util.ObjectChecker.requireNonNullAndWrapUnmodifiable;

import java.util.List;

public final class HolonomicPath extends Path {
    public final List<? extends HolonomicWaypoint> holonomicWaypoints;

    @SuppressWarnings("unchecked")
    public HolonomicPath(List<? extends HolonomicWaypoint> holonomicWaypoints, List<Obstacle> obstacles) throws NullPointerException {
        super(requireNonNullAndWrapUnmodifiable(holonomicWaypoints,
                "List of holonomic waypoints cannot be null", "Holonomic waypoint cannot be null"),
                obstacles);

        this.holonomicWaypoints = (List<? extends HolonomicWaypoint>) waypoints;
    }

    @Override
    public String toString() {
        return holonomicWaypoints.toString();
    }
}
